### PR TITLE
Ft/seller registration 

### PR DESCRIPTION
**Pull request by adisa39**
backend integration & UI revamped
Modified authentication logics to remove local storage

### CODE REVIEW

**Comment by swoocn** [comment]
Hey @adisa39 - This is what I see in the sandbox environment after I synced your frontend and backend changes; can we handle the exception or otherwise? Looks like the login worked though! ->

image

**Comment by adisa39** [comment]
> Hey @adisa39 - This is what I see in the sandbox environment after I synced your frontend and backend changes; can we handle the exception? Looks like the login worked though! ->
> 
> image

Yes I think that is from PiSDK onPaymentComplete function which must be passed as parameter in PiSDK user authentication function which is yet to be implemented. You can refer to PiSDK doc for details I think and necessary modifications

File: context/AppContextProvider.tsx [hunk 1]
```diff
@@ -39,13 +40,14 @@ interface AppContextProviderProps {
...
-  const [currentUser, setCurrentUser] = useState<IUser | null>(null);
+  const [currentUser, setCurrentUser] = useState<string | null>(null);
 
   const registerUser = async () => {
     const isInitiated= await Pi.initialized;
+
     if (isInitiated) {  
       try {
-        localStorage.removeItem("mapOfPiToken");
+        // localStorage.removeItem("mapOfPiToken");
```
Comment (swoocn) [inline context/AppContextProvider.tsx]: No need to comment lines; Please just remove.  We can always refer and/ or recover if needed through the commit history.

File: src/app/[locale]/seller/registration/page.tsx [hunk 2]
```diff
@@ -194,18 +210,63 @@ const SellerRegistrationForm = () => {
   };
 
   return (
-    <>
+    <>    
+      {loading && <div className="loading">Loading...</div>}
```
Comment (swoocn) [inline src/app/[locale]/seller/registration/page.tsx]: Is this loading section necessary when we have a skeleton preload effect?

File: src/app/[locale]/seller/registration/page.tsx [hunk 3]
```diff
@@ -227,58 +288,7 @@ const SellerRegistrationForm = () => {
             }}
           />
         </Link>
-        <div className="mb-4 mt-3 ml-auto w-min">
```
Comment (swoocn) [inline src/app/[locale]/seller/registration/page.tsx]: Not sure why the collapsible elements were removed.  Please revert and restore them.  Thanks!

File: src/app/[locale]/seller/reviews/[id]/page.tsx [hunk 4]
```diff
@@ -50,7 +50,7 @@ function SellerReviews({
             date: resolveDate(feedback.review_date).date,
             time: resolveDate(feedback.review_date).time,
             user: feedback.review_giver_id,
-            reviewId: feedback.review_id,
+            reviewId: feedback._id,
```
Comment (swoocn) [inline src/app/[locale]/seller/reviews/[id]/page.tsx:54]: 👍

File: src/services/reviewsApi.ts [hunk 5]
```diff
@@ -25,10 +26,21 @@ export const fetchReviews = async (sellerId:string) => {
...
-    const response = await axiosClient.post('/review-feedback/add', formData, );
-    return response.data;
-  } catch (error) {
-    console.error('Error creating review: ', error);
+    const response = await axiosClient.post('/review-feedback/add', formData);
+    if (response.status === 200) {
+      return response.data;
+    }
+
+    if (response.status === 400) {
```
Comment (swoocn) [inline src/services/reviewsApi.ts]: I don’t think it’s necessary to send all these messages back to the user. Instead, we could simplify it to two messages: one to confirm that the review was successfully added and another to indicate if the review submission failed. We can log the detailed errors on our end.

File: src/app/[locale]/seller/registration/page.tsx (hunk 2 above)
Comment (adisa39) [reply src/app/[locale]/seller/registration/page.tsx]: Yes It is cos the page is loaded first before the data except we are to use partial loading method

File: src/app/[locale]/seller/registration/page.tsx (hunk 3 above)
Comment (adisa39) [reply src/app/[locale]/seller/registration/page.tsx]: Yeah that will be reversed

**Review by swoocn — APPROVED** [review]
Approved after additional adjustments; refactor + cleanup. 👍

**Pull request merged**